core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/PathTrackingInputFormat.java

```diff
@@ -41,14 +47,20 @@
   private static final String PATH_FIELD = "path.tracking.path.field";
   private static final String FILENAME_ONLY = "path.tracking.filename.only";
 
+  private static String fileSourceFormat = "text";
```

Comment:
don't store static state like this.

Configuration options are set in the Configuration object, then mapreduce takes care of serializing and distributing that Configuration to all mappers/reducers. The process that calls configure will not be the same process that calls createRecordReader.